test/helpers/helpers.ts

```diff
@@ -37,3 +38,53 @@ export const scrollToElement = (element) => {
 export const waitForPresenceOfElement = (element) => {
   browser.wait(ExpectedConditions.presenceOf(element));
 };
+
+export const resetTestDataState = (username) => {
+  waitForOverlay('click-block-active');
+  clickBackButton();
+  // Logout
+  logout();
+  // Login
+  logInToApplication(TEST_CONFIG.users[username].username, TEST_CONFIG.users[username].password);
+  // Refresh application
+  loadApplication().then(() => {
+  // Small wait to make sure the action has initiated
+    browser.driver.sleep(TEST_CONFIG.ACTION_WAIT);
+  });
+};
+
+export const clickBackButton = () => {
+  const backButton = getElement(by.xpath('//page-journal//button//span[text()="Back"]'));
+  clickElement(backButton);
+};
+
+export const getElement = (elementBy) => {
+  const foundElement = element(elementBy);
+  browser.wait(ExpectedConditions.presenceOf(foundElement));
+  return foundElement;
+};
+
+/**
+ * A framework safe click method.
+ * @param fieldElement the element to click
+ */
+export const clickElement = (fieldElement) => {
+  browser.wait(ExpectedConditions.elementToBeClickable(fieldElement));
+  fieldElement.click().then((promise) => {
+    return isReady(promise);
+  });
+};
+
+/**
+ * Checks whether the page is ready to be interacted with.
+ * Ionic adds an overlay preventing clicking until the page is ready so we need to wait for that to disappear.
+ * @param promise the promise to return when the page is ready
+ */
+export const isReady = (promise) => {
+  // There is a 200ms transition duration we have to account for
+  browser.sleep(TEST_CONFIG.ACTION_WAIT);
+  // Then wait for the page to become active again
+  browser.wait(ExpectedConditions.stalenessOf(element(by.className('click-block-active'))));
+  // Then return the original promise
+  return promise;
+};
```

Comment:
Hate the filename - helpers within a folder called helpers seems unnecesary - feel the file should be something more descriptive like domHelpers, or interactionHelpers

Reply:
removed the hate